### ISSUE TITLE
Unacceptable framework overhead for huge networks

### ISSUE THREAD

**Issue by IFeelBloated** [tensorflow/tensorflow#23545]
I posted the same issue for Keras [here](https://github.com/keras-team/keras/issues/11592), but I guess it's more of a tensorflow problem in general.
so for a large network with thousands or even more layers (the network might not be that large mathematically since each layer could contain just a small number of parameters, but u get the idea), the overhead of basically anything messing with some graph stuff becomes unacceptable, things like model compilation, XLA optimization, gradient checkpoint, and also the keras "save_weights" functions are of ABYSMAL performance and could take literally DAYS!! even longer than the actual training process on GPU, I noticed that every time something like the aforementioned stuff was being executed, the python script just got stuck, one CPU core skyrocketed to 100% usage and the rest 39 cores were just almost 0% usage, then it would take hours and hours before that thing was done and then we could finally move on to the actual training.
It's honestly just frustrating and unacceptable, any idea to solve this?
sample script to reproduce the performance problem: https://gist.github.com/IFeelBloated/6e85b251f66941b7eb50f987c95ce69b

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ezhulenev: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gowthamkpr** [comment]
@IFeelBloated As mentioned [here](https://github.com/tensorflow/tensorflow/pull/23596#issuecomment-454192185), this is not possible. Please let me know. Thanks!
